devito/passes/iet/openmp.py

```diff
@@ -135,7 +135,20 @@ def _make_clauses(cls, ncollapse=None, chunk_size=None, nthreads=None,
             clauses.append('num_threads(%s)' % nthreads)
 
         if reduction:
-            args = ','.join(str(i) for i in reduction)
+            reds = []
+            for i in reduction:
+                # OMP expects a size not an index as input of reduction, such as
+                # reduction(+:f[1:3])
+                if i.is_Indexed:
+                    bounds = [(k,) if k.is_Number else
```

Comment:
I don't think we need this `if k.is_Number` thing ...

```
[(k.subs(d, d.symbolic_min), k.subs(d, d.symbolic_max + 1)) for k, d, ...]
```

should suffice?

Reply:
May need extra `sympyify` to have it to work like that but yes.

Reply:
Atually nom you need the `if k.is_Number` or you'd end up with `(k, k)` for the single index case instead of just `(k,)`